.github/workflows/package-generate-repository.yaml

```diff
@@ -0,0 +1,113 @@
+name: Package - Generate Package Repository
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - addons/packages/**/*/package.yaml
+      - addons/packages/**/*/metadata.yaml
+      - addons/repos/main.yaml
+  pull_request:
+    branches:
+      - "*"
```

Comment:
@dvonthenen Maybe I need some help here.

To be honest, I would like for this workflow to be run for any branch, on any fork. However, we can't have that because secrets are not shared with forks, and you need (secret) credentials to push to our registry.

So with that constraint, I want the workflow to run anytime a merge to main, or a release branch, happens. There are always updates to packages, and I would not want to restrict this workflow from  triggering only when a release is being cut. We've always stated that package changes happen out of band to our release cycle.

Also, this workflow creates repos on the staging registry. I am going to propose another workflow that will promote from staging to production at a later date.

Reply:
@dvonthenen Looking at this again... This section is for Pull Requests. In theory, all PR's should come from Forks of the repo. As I've stated in my previous comment, this workflow will not run on Forks due to secrets not being shared. When a PR is merged to main (whether or not the PR is on a fork), the workflow will trigger.

So it's feeling like the `pull_request` stanza should just be removed?

Reply:
I deleted the `pull_request` stanza.

Reply:
Yea, if we are just doing it when merging to main of the release branch... it should probably look like this:
https://github.com/vmware-tanzu/community-edition/blob/main/.github/workflows/check-all.yaml#L2-L8

Reply:
@dvonthenen I updated the workflow to match the `on` stanza of your example. I really only care about merges to main, but there shouldn't be any issues with doing this for release branches as well.